tests/testthat/test-prepare_matches_mapper.R

```diff
@@ -1,9 +1,36 @@
 test_that("`multi_match` column has either `TRUE` or `FALSE` values", {
   out <- prepare_matches_mapper(small_matches_mapper, ecoinvent_activities)
-  expect_equal(unique(out$multi_match), c("TRUE", "FALSE"))
+  expect_equal(unique(out$multi_match), c(TRUE, FALSE))
 })
 
 test_that("`completion` column has either `low`, `medium`, or `high` values", {
   out <- prepare_matches_mapper(small_matches_mapper, ecoinvent_activities)
   expect_equal(unique(out$completion), c("low", "high", "medium"))
 })
+
+test_that("`prepare_matches_mapper` doesn't have any empty string", {
+  mm <- tibble_names("a", names(matches_mapper))
+  mm$activity_uuid_product_uuid <- NA_character_
+  ea <- tibble_names("a", names(ecoinvent_activities))
+
+  out <- prepare_matches_mapper(mm, ea)
+  expect_equal(out$activity_uuid_product_uuid, NA_character_)
+})
+
+test_that("`multi_match` column outputs `FALSE` for null and empty string value", {
+  mm <- tibble(
+    country = c("x", "y"),
+    main_activity = c("a", "a"),
+    clustered = c("a", "a"),
+    ep_id = c("a", "a"),
+    activity_uuid_product_uuid = c("a", "a"),
+    multi_match = c(NA_character_, ""),
+    completion = c("a", "a"),
+    category = c("a", "a")
+  )
```

Comment:
This test doesn't seem correct. In the dataset `matches_mapper` the column `multi_match` is logical.

``` r
is.logical(tiltIndicatorAfter::matches_mapper$multi_match)
#> [1] TRUE
```

We could enforce this with an assertion, so we're sure it's never a character.

```r
prepare_matches_mapper <- function(mapper, activities) {
  stopifnot(is.logical(mapper$multi_match))
  ...
}
```

Then we don't need this line:

```r
   ...
    mutate(multi_match = as.logical(.data$multi_match)) |>
```

But I'm getting a bit lost and may have lost track of how we arrived here :-)

Reply:
@maurolepore Indeed you are right that the `multi_match` column should remain logical. However, this line of code change the type of this column:
```summarise(across(everything(), ~ paste0(na.omit(unique(.)), collapse = "; ")), .by = c("country", "main_activity", "clustered"))```

After analysing the real data, I identified that the requirement to run this command is only required for these two columns `activity_uuid_product_uuid` and `activity_name`. Hence, I have applied a new check to ensure that the need to run this command on other columns will not change in the future (by stoping the code from running). Please let me know if the new logic make sense to you! We can have a call this time to align easily if you'd like :)